autorest/codegen/serializers/builder_serializer.py

```diff
@@ -31,9 +31,14 @@
 OrderedSet = Dict[T, None]
 
 
+# "str (optional)" => "str" # (optional)
+def _improve_template(template):
```

Comment:
@msyyc I would not make the change here. The change should be made in primitive schema's `_add_optional_and_default_value_template_representation`, not afterwards